.github/actions/javascript/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js

```diff
@@ -30,122 +25,115 @@ function getDeployTableMessage(platformResult) {
-const desktopResult = getDeployTableMessage(core.getInput('DESKTOP', {required: true}));
-const iOSResult = getDeployTableMessage(core.getInput('IOS', {required: true}));
-const webResult = getDeployTableMessage(core.getInput('WEB', {required: true}));
-
-const workflowURL = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`;
-
-/**
- * @param {String} deployer
- * @param {String} deployVerb
- * @param {String} prTitle
- * @returns {String}
- */
-function getDeployMessage(deployer, deployVerb, prTitle) {
-    let message = `🚀 [${deployVerb}](${workflowURL}) to ${isProd ? 'production' : 'staging'}`;
-    message += ` by https://github.com/${deployer} in version: ${version} 🚀`;
-    message += `\n\n platform | result \n ---|--- \n🤖 android 🤖|${androidResult} \n🖥 desktop 🖥|${desktopResult}`;
-    message += `\n🍎 iOS 🍎|${iOSResult} \n🕸 web 🕸|${webResult}`;
-
-    if (deployVerb === 'Cherry-picked' && !/no qa/gi.test(prTitle)) {
-        // eslint-disable-next-line max-len
-        message +=
-            '\n\n@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.';
-    }
-
-    return message;
-}
-
 /**
  * Comment Single PR
  *
  * @param {Number} PR
  * @param {String} message
  * @returns {Promise<void>}
  */
-function commentPR(PR, message) {
-    return GithubUtils.createComment(context.repo.repo, PR, message)
-        .then(() => console.log(`Comment created on #${PR} successfully 🎉`))
-        .catch((err) => {
-            console.log(`Unable to write comment on #${PR} 😞`);
-            core.setFailed(err.message);
-        });
+async function commentPR(PR, message) {
+    try {
+        await GithubUtils.createComment(context.repo.repo, PR, message);
+        console.log(`Comment created on #${PR} successfully 🎉`);
+    } catch (err) {
+        console.log(`Unable to write comment on #${PR} 😞`);
+        core.setFailed(err.message);
+    }
 }
 
-const run = function () {
+const workflowURL = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`;
+
+async function run() {
+    const prList = _.map(ActionUtils.getJSONInput('PR_LIST', {required: true}), (num) => Number.parseInt(num, 10));
+    const isProd = ActionUtils.getJSONInput('IS_PRODUCTION_DEPLOY', {required: true});
+    const version = core.getInput('DEPLOY_VERSION', {required: true});
+
+    const androidResult = getDeployTableMessage(core.getInput('ANDROID', {required: true}));
+    const desktopResult = getDeployTableMessage(core.getInput('DESKTOP', {required: true}));
+    const iOSResult = getDeployTableMessage(core.getInput('IOS', {required: true}));
+    const webResult = getDeployTableMessage(core.getInput('WEB', {required: true}));
+
+    /**
+     * @param {String} deployer
```

Comment:
Sorry, I thought i had posted this already but then I saw that I never submitted the review 😓

Reply:
Not really because it uses `iOSResult`, `androidResult`, etc... and those are all defined inside `run`. They need to be defined inside run just so that we can mock `core.getInput` and have it re-compute when we re-run. At least, that was how I was able to get the tests working correctly...

Reply:
Ah that makes sense. Thanks for clarifying!